dipy/align/tests/test_mattes.py

```diff
@@ -0,0 +1,776 @@
+        for min_int in [-10.0, 0.0, 10.0]:
+            for intensity_range in [0.1, 1.0, 10.0]:
+                factor = 1
+                max_int = min_int + intensity_range
+                M = MattesBase(nbins)
+                # Make a pair of 4-pixel images, introduce +/- 1 values
+                # that will be excluded using a mask
+                static = np.array([min_int - 1.0, min_int,
+                                   max_int, max_int + 1.0])
+                # Multiply by an arbitrary value (make the ranges different)
+                moving = factor * np.array([min_int, min_int - 1.0,
+                                  max_int + 1.0, max_int])
+                # Create a mask to exclude the invalid values (beyond min and
+                # max computed above)
+                static_mask = np.array([0, 1, 1, 0])
+                moving_mask = np.array([1, 0, 0, 1])
+
+                M.setup(static, moving, static_mask, moving_mask)
+
+                # Test bin_normalize_static at the boundary
+                normalized = M.bin_normalize_static(min_int)
+                assert_almost_equal(normalized, M.padding)
+                index = M.bin_index(normalized)
+                assert_equal(index, M.padding)
+                normalized = M.bin_normalize_static(max_int)
+                assert_almost_equal(normalized, nbins - M.padding)
+                index = M.bin_index(normalized)
+                assert_equal(index, nbins - 1 - M.padding)
+
+                # Test bin_normalize_moving at the boundary
+                normalized = M.bin_normalize_moving(factor * min_int)
+                assert_almost_equal(normalized, M.padding)
+                index = M.bin_index(normalized)
+                assert_equal(index, M.padding)
+                normalized = M.bin_normalize_moving(factor * max_int)
+                assert_almost_equal(normalized, nbins - M.padding)
+                index = M.bin_index(normalized)
+                assert_equal(index, nbins - 1 - M.padding)
+
+                # Test bin_index not at the boundary
+                delta_s = (max_int - min_int)/(nbins - 2*M.padding)
+                delta_m = factor * (max_int - min_int)/(nbins - 2*M.padding)
+                for i in range(nbins - 2*M.padding):
+                    normalized = M.bin_normalize_static(min_int +
+                                                        (i+0.5)*delta_s)
+                    index = M.bin_index(normalized)
+                    assert_equal(index, M.padding + i)
+
+                    normalized = M.bin_normalize_moving(factor*min_int +
+                                                        (i+0.5)*delta_m)
+                    index = M.bin_index(normalized)
+                    assert_equal(index, M.padding + i)
+
+
+def test_mattes_densities_dense():
+    # Test the computation of the joint intensity distribution
+    # using a dense set of values (from a pair of images)
+    np.random.seed(1246592)
```

Comment:
Oh! of course! another remanent of the original tests (previously the code in `create_random_image_pair` was part of the test function, and then I refactored it). Now passing the seed as argument.